goal_src/levels/common/battlecontroller.gc

```diff
@@ -5,3 +5,1043 @@
+     (set! (-> a1-7 num-params) 0)
+     (set! (-> a1-7 message) 'spawn)
+     (let ((t9-1 send-event-function)
+           (v1-3 (-> v1-2 trigger-actor))
+           )
+      (t9-1 (if v1-3
+             (-> v1-3 extra process)
+             )
+       a1-7
+       )
+      )
+     )
+    (battlecontroller-spawn-creature-at-spawner gp-0 0)
+    )
+   )
+  0
+  (none)
+  )
+
+(defbehavior battlecontroller-fill-all-spawners battlecontroller ()
+  (dotimes (gp-0 (-> self spawner-count))
+   (battlecontroller-spawn-creature-at-spawner gp-0 1)
+   )
+  (when (nonzero? (-> self path-spawn))
+   (let ((gp-1 (new 'stack-no-clear 'vector)))
+    (dotimes (s5-0 (-> self path-spawn curve num-cverts))
+     (eval-path-curve-div! (-> self path-spawn) gp-1 (the float s5-0) 'interp)
+     (let ((s4-0 (battlecontroller-spawn-creature gp-1 (-> self root trans))))
+      (when (handle->process s4-0)
+       (suspend)
+       (let ((a1-4 (new 'stack-no-clear 'event-message-block)))
+        (set! (-> a1-4 from) self)
+        (set! (-> a1-4 num-params) 0)
+        (set! (-> a1-4 message) 'cue-chase)
+        (send-event-function (handle->process s4-0) a1-4)
+        )
+       )
+      )
+     )
+    )
+   )
+  0
+  (none)
+  )
+
+(defbehavior battlecontroller-battle-begin battlecontroller ()
+  (set-setting! *setting-control* self 'music 'danger 0.0 0)
+  (let ((gp-0 (entity-actor-count (-> self entity) 'kill-actor)))
+   (dotimes (s5-0 gp-0)
+    (let ((v1-2 (entity-actor-lookup (-> self entity) 'kill-actor s5-0)))
+     (if v1-2
+      (logior! (-> v1-2 extra perm status) (entity-perm-status bit-9))
+      )
+     )
+    )
+   )
+  (let ((gp-1 (entity-actor-count (-> self entity) 'trigger-actor)))
+   (dotimes (s5-1 gp-1)
+    (let ((s4-0 (new 'stack-no-clear 'event-message-block)))
+     (set! (-> s4-0 from) self)
+     (set! (-> s4-0 num-params) 0)
+     (set! (-> s4-0 message) 'trigger)
+     (let ((s3-0 send-event-function)
+           (v1-7 (entity-actor-lookup (-> self entity) 'trigger-actor s5-1))
+           )
+      (s3-0 (if v1-7
+             (-> v1-7 extra process)
+             )
+       s4-0
+       )
+      )
+     )
+    )
+   )
+  (level-hint-spawn
+   (game-text-id zero)
+   (the-as string #f)
+   (-> self entity)
+   *entity-pool*
+   (game-task none)
+   )
+  0
+  (none)
+  )
+
+(defbehavior battlecontroller-off battlecontroller ()
+  (clear-pending-settings-from-process *setting-control* self 'music)
+  (let ((gp-0 (entity-actor-count (-> self entity) 'kill-actor)))
+   (dotimes (s5-0 gp-0)
+    (let ((v1-2 (entity-actor-lookup (-> self entity) 'kill-actor s5-0)))
+     (if v1-2
+      (logclear! (-> v1-2 extra perm status) (entity-perm-status bit-9))
+      )
+     )
+    )
+   )
+  0
+  (none)
+  )
+
+(defbehavior battlecontroller-battle-end battlecontroller ()
+  (case (-> (level-get-target-inside *level*) name) 
+   (('citadel)
+     (level-hint-spawn
+      (game-text-id misty-battle-finished)
+      "sksp0378"
```

Comment:
do you know what this says?

Reply:
"the doors open, lets get going"

Reply:
It's under `citadel` in the case statement - could this be for the room with all the annoying hopping enemies?